src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```diff
@@ -47,33 +47,37 @@ using namespace Gui;
 /* TRANSLATOR PartDesignGui::TaskRevolutionParameters */
 
 TaskRevolutionParameters::TaskRevolutionParameters(PartDesignGui::ViewProvider* RevolutionView, QWidget *parent)
-    : TaskSketchBasedParameters(RevolutionView, parent, "PartDesign_Revolution", tr("Revolution parameters"))
-    , ui(new Ui_TaskRevolutionParameters)
+    : TaskSketchBasedParameters(RevolutionView, parent, "PartDesign_Revolution", tr("Revolution parameters")),
+      ui(new Ui_TaskRevolutionParameters), proxy(new QWidget(this))
 {
     // we need a separate container widget to add all controls to
-    proxy = new QWidget(this);
     ui->setupUi(proxy);
     QMetaObject::connectSlotsByName(this);
     this->groupLayout()->addWidget(proxy);
 
-    //bind property mirrors
-    PartDesign::ProfileBased* pcFeat = static_cast<PartDesign::ProfileBased*>(vp->getObject());
+    // bind property mirrors
+    auto *pcFeat = dynamic_cast<PartDesign::ProfileBased*>(vp->getObject());
+    if (!pcFeat)
+        return;

```

Comment:
Returning here is a very bad way because it leaves a half-initialized object. Instead an exception must raised so that the compiler cleans it up again.

But because further down the actual type checks are done the lines 59-61 can be replaced with: `auto* pcFeat = vp->getObject()`

Reply:
I se but the whole pcFeat is not necessary at all. I removed it therefore now.